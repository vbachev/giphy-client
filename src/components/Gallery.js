import React from 'react'
import Image from './Image'

// @TODO:
// - fix mobile layout

const Gallery = ({ images, keyword, totalImages, layout, onLayoutChange }) => {
  return (
    <div className={`images-container layout-${layout}`}>

      <div className='images-layout'>
        Display as:
        <button className='layout-button layout-grid' onClick={() => onLayoutChange('grid')}>
          Grid
        </button>
        <button className='layout-button layout-list' onClick={() => onLayoutChange('list')}>
          List
        </button>
      </div>

      <div className='images-summary'>
        <span className='images-keyword'>
          {keyword}
        </span>
        <span className='images-total'>
          {totalImages} GIFs
        </span>
      </div>

      <ul className='images-list'>
        {images.map(image => (
          <li key={image.title}>
            <Image {...image} />
          </li>
        ))}
      </ul>
    </div>
  )
}

export default Gallery
