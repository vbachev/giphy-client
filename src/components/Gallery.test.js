import React from 'react'
import renderer from 'react-test-renderer'
import Gallery from './Gallery'

const mockImages = [
  { id: 'foo', url: 'https://foo.com', title: 'foo' },
  { id: 'bar', url: 'https://bar.com', title: 'bar' }
]

test('it renders a gallery', () => {
  const component = renderer.create(
    <Gallery images={mockImages} keyword='foo' totalImages={23} layout='grid' onLayoutChange={() => {}} />
  )
  expect(component.toJSON()).toMatchSnapshot()
})
