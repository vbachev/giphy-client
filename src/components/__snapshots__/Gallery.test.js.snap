// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a gallery 1`] = `
<div
  className="images-container layout-grid"
>
  <div
    className="images-layout"
  >
    Display as:
    <button
      className="layout-button layout-grid"
      onClick={[Function]}
    >
      Grid
    </button>
    <button
      className="layout-button layout-list"
      onClick={[Function]}
    >
      List
    </button>
  </div>
  <div
    className="images-summary"
  >
    <span
      className="images-keyword"
    >
      foo
    </span>
    <span
      className="images-total"
    >
      23
       GIFs
    </span>
  </div>
  <ul
    className="images-list"
  >
    <li>
      <div
        className="image-container"
      >
        <img
          alt="foo"
          className="image"
          src="https://foo.com"
        />
      </div>
    </li>
    <li>
      <div
        className="image-container"
      >
        <img
          alt="bar"
          className="image"
          src="https://bar.com"
        />
      </div>
    </li>
  </ul>
</div>
`;
